tools/mininet/tofino-model.py

```diff
@@ -0,0 +1,253 @@
+            self.stratumProc_h = None
+        Switch.stop(self, deleteIntfs)
+
+    def start(self, controllers):
+
+        if not self.stopped: return
+
+        #TODO this could prob be replaced by setting an ip in the parent.init
+        # or something to trigger ip link up...
+        self.cmd("/usr/sbin/ip l set dev lo up")
+
+        with open(self.chassisConfigFile, 'w') as fp:
+            fp.write(self.getChassisConfig())
+        with open(self.portVethFile, 'w') as fp:
+            fp.write(self.getPortVethMap())
+        tof_cmd_string = " ".join(
+            [
+                TOFINO_MODEL_BIN,
+                '--p4-target-config=/etc/stratum/tofino_skip_p4.conf',
+                f'-f{self.portVethFile}'
+            ]
+        )
+        stratum_cmd_string = " ".join(
+            [
+                STRATUM_BIN,
+                f'-chassis_config_file={self.chassisConfigFile}',
+                '-bf_sim',
+                '-enable_onlp=false',
+                '-bf_switchd_background=false',
+                '-v=6',
+                '-alsologtostderr=true',
+                f'-bf_sde_install={self.sdeInstall}',
+                '-bf_switchd_cfg=/etc/stratum/tofino_skip_p4.conf',
+            ]
+        )
+
+        try:
+            # Write cmd_string to log for debugging.
+            self.t_logfd = open(f'{self.tmpDir}/tofino_model_process.log' , "w")
+            self.t_logfd.write(tof_cmd_string + "\n\n" + "-" * 80 + "\n\n")
+            self.t_logfd.flush()
+            self.s_logfd = open(f'{self.tmpDir}/stratum_process.log', "w")
+            self.s_logfd.write(stratum_cmd_string + "\n\n" + "-" * 80 + "\n\n")
+            self.s_logfd.flush()
+
+            self.t_modelProc_h = self.popen(tof_cmd_string, stdout=self.t_logfd, stderr=self.t_logfd)
+            self.stratumProc_h = self.popen(stratum_cmd_string, stdout=self.s_logfd, stderr=self.s_logfd)
+
+            # We want to be notified if processes quits prematurely...
+            self.stopped = False
+            threading.Thread(target=watchdog, args=[self]).start()
+
+        except Exception:
+            StratumTofinoModel.mininet_exception = 1
+            self.stop()
+            raise
+
+    def getPortVethMap(self):
+        intf_number = 1
+        portsVeth = list()
+        for intf_name in self.intfNames():
+            if intf_name == 'lo': continue
+            portsVeth.append(
+                {
+                    "device_port": intf_number,
+                    "veth1": int(intf_name[4:]),
+                    "veth2": int(intf_name[4:])+100
+                }
+            )
+            intf_number+=1
+
+        data =  { "PortToVeth": portsVeth }
+        return json.dumps(data, indent=4)+"\n"
+
+    def getChassisConfig(self):
+        config = """description: "chassis config bf tofino model {name}"
+chassis {{
+  platform: PLT_GENERIC_BAREFOOT_TOFINO
+  name: "{name}"
+}}
+nodes {{
+  id: {nodeId}
+  name: "{name} node {nodeId}"
```

Comment:
I think we can remove node id since we can find it from the `id` field. any reason we need it in the `name` field?